tests/components/vallox/test_config_flow.py

```diff
@@ -0,0 +1,242 @@
+
+
+async def test_form_os_error_cannot_connect(hass: HomeAssistant) -> None:
+    """Test that cannot connect error is handled."""
+    init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch(
+        "vallox_websocket_api.Vallox.get_info",
+        side_effect=OSError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            init["flow_id"],
+            {"host": "5.6.7.8"},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"host": "cannot_connect"}
+
+
+async def test_form_already_configured(hass: HomeAssistant) -> None:
+    """Test that already configured error is handled."""
+    init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.vallox.config_flow.ConfigFlow.host_already_configured",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            init["flow_id"],
+            {"host": "20.40.10.30"},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"host": "already_configured"}
+
+
+async def test_import(hass: HomeAssistant) -> None:
+    """Test that import is handled."""
+    name = "Vallox 90 MV"
+
+    with patch(
+        "vallox_websocket_api.Vallox.get_info",
+        return_value={"model": "Vallox 90 MV"},
+    ), patch(
+        "homeassistant.components.vallox.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={"host": "1.2.3.4", "name": name},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == name
+    assert result["data"] == {"host": "1.2.3.4", "name": "Vallox 90 MV"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_invalid_host(hass: HomeAssistant) -> None:
+    """Test that invalid host error is handled during import."""
+    name = "Vallox 90 MV"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data={"host": "vallox90mv.&host.name", "name": name},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "invalid_host"
+
+
+async def test_import_already_configured(hass: HomeAssistant) -> None:
+    """Test that an already configured Vallox device is handled during import."""
+    name = "Vallox 145 MV"
+
+    with patch(
+        "homeassistant.components.vallox.config_flow.ConfigFlow.host_already_configured",
+        return_value=True,
+    ):
```

Comment:
Don't patch the integration.  Instead create a MockConfigEntry with the same host and verify the config flow aborts

Reply:
👍 

I think I got this done, please verify that I got it right.